src/main/java/com/bookingTour/controller/CategoryController.java

```diff
@@ -61,21 +61,31 @@ public String add(Locale locale, Model model) {
     public String create(@ModelAttribute("category") @Validated CategoryInfo categoryInfo, BindingResult bindingResult, Model model, HttpServletRequest request) throws Exception {
         if (bindingResult.hasErrors()) {
             logger.info("Returning add category page, validate failed");
-            return "categories/create";
+            return "templates/error";
         }
         categoryService.addCategory(categoryInfo);
         return "redirect: " + request.getContextPath() + "/categories";
     }
 
     @GetMapping(value = "/{id}")
     public String show(@PathVariable Long id, Model model) {
-        model.addAttribute("category", categoryService.findCategory(id));
+    	CategoryInfo category = categoryService.findCategory(id);
+    	if (category == null) {
+    		model.addAttribute("errorMsg", "Category ID not found");
```

Comment:
`Category ID not found ` cái này chỉ cần nói category not found thôi, ko cần nói Id nhé, vì end-user có hiểu id là cái gì đâu :p